sdks/python/apache_beam/io/fileio.py

```diff
@@ -139,12 +139,23 @@ def close(self, file_handle):
   @check_accessible(['file_path_prefix', 'file_name_suffix'])
   def initialize_write(self):
     file_path_prefix = self.file_path_prefix.get()
-    file_name_suffix = self.file_name_suffix.get()
-    tmp_dir = file_path_prefix + file_name_suffix + time.strftime(
-        '-temp-%Y-%m-%d_%H-%M-%S')
+
+    tmp_dir = self._create_temp_dir(file_path_prefix)
     FileSystems.mkdirs(tmp_dir)
     return tmp_dir
 
+  def _create_temp_dir(self, file_path_prefix):
+    base_path, last_component = FileSystems.split(file_path_prefix)
+    if not last_component:
+      raise ValueError('Cannot create a temporary directory for root path'
```

Comment:
So the idea was to remove the inconsistency of supporting root paths in only some of the cases (for any file-system). So I don't think this change is GCS specific. I think doing more advanced checks here (such as writability) is overkill. Also any other file-system that support writing to arbitrary top-level directories should produce appropriate results from split() method. For example for file-system ABC split('abc://xyz') can result in ('abc://', 'xyz').

Note that This does not prevent local file-systems from writing to top level directories BTW since os.split('/tmp') ends up being ('/', 'tmp').